src/Std/Time/DateTime/PlainDateTime.lean

```diff
@@ -0,0 +1,467 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Date
+import Std.Time.Time
+import Std.Time.Internal
+import Std.Time.DateTime.Timestamp
+
+namespace Std
+namespace Time
+open Internal
+
+set_option linter.all true
+
+/--
+Represents a date and time with components for Year, Month, Day, Hour, Minute, Second, and Nanosecond.
+-/
+structure PlainDateTime where
```

Comment:
I think it would be cool if there were functions like `PlainDate.atTime : PlainDate -> PlainTime -> PlainDateTime` and also `PlainDateTime.withTime : PlainDateTime -> PlainTime -> PlainDateTime`. Of course, these are merely wrapper around the constructor/the `with` syntax, but I've been wishing these functions were there while playing with the library.

Reply:
Added!